jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerFactory.java

```diff
@@ -0,0 +1,213 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.logging;
+
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Consumer;
+
+import org.slf4j.ILoggerFactory;
+import org.slf4j.Logger;
+
+public class JettyLoggerFactory implements ILoggerFactory
+{
+    protected static JettyLoggerFactory getLoggerFactory()
```

Comment:
Do we need this static instance at all?  The instance we will use is the one created by the `JettyLoggingServiceProvider` so why confuse things and have another lazy static?

Reply:
That static is used by both our implementation (to handle cascading level and hidestacks runtime changes) and the JettyLoggingServiceProvider's own `initialize()` phase.

Reply:
I think this is needlessly complex and the use of initiailize methods means we cannot use fast finals.
The JettyLoggingServiceProvider should just make it's own instance and pass the configuration into the constructor. If slf4j is providing us with a life cycle mechanism, then I don't see why we need to duplicate with statics.  Can you try to get rid of this

Reply:
Removed singletons at JettyLoggerFactory